src/bytes_mut.rs

```diff
@@ -733,9 +733,10 @@ impl BytesMut {
 
     /// Absorbs a `BytesMut` that was previously split off.
     ///
-    /// If the two `BytesMut` objects were previously contiguous, i.e., if
-    /// `other` was created by calling `split_off` on this `BytesMut`, then
-    /// this is an `O(1)` operation that just decreases a reference
+    /// If the two `BytesMut` objects were previously contiguous and not mutated
+    /// in a way that causes re-allocation.
+    /// i.e., if `other` was created by calling `split_off` on this `BytesMut`,
+    /// then this is an `O(1)` operation that just decreases a reference
```

Comment:
This sentence seems a bit broken - there's a period in the middle of a sentence.